workflow/rules/quant.smk

```diff
@@ -23,6 +23,11 @@ rule kallisto_quant:
         extra=kallisto_params,
     conda:
         "../envs/kallisto.yaml"
+    # around 4 gb of memory usage with 1 thread. (hg38)
+    # over 8gb peak memory usage with 8 threads. (hg38)
+    threads: 8
+    resources:
+        mem_mb=10000,
     shell:
         "kallisto quant -i {input.idx} -o {output} "
-        "{params.extra} {input.fq} 2> {log}"
+        "{params.extra} -t {threads} {input.fq} 2> {log}"
```

Comment:
While we are at it, I'd suggest moving to using the respective wrapper. This makes maintenance easier in the long run:
https://snakemake-wrappers.readthedocs.io/en/stable/wrappers/kallisto/quant.html

Same for the `rule kallisto_index` above:
https://snakemake-wrappers.readthedocs.io/en/stable/wrappers/kallisto/index.html

Reply:
Should be resolved! Please review @dlaehnemann